ElementX/Sources/Other/VoiceMessage/WaveformProgressCursorModifier.swift

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Compound
+import SwiftUI
+
+extension View {
+    func waveformProgressCursor<CursorView: View>(progress: CGFloat,
```

Comment:
```suggestion
    func progressCursor<CursorView: View>(progress: CGFloat,
```

This isn't now really linked to waveforms, it can be used anywhere! ;)